### PR TITLE
Add Datadog metrics sample project

### PR DESCRIPTION
**Pull request by pnevmap**

### CODE REVIEW

File: samples/logging/datadog/sample.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Capture and visualize metrics using Datadog
+summary: Illustrates how to capture, store and visualize NServiceBus metrics in Datadog.
+component: Metrics
+isLearningPath: true
```
Comment (ramonsmits) [inline samples/logging/datadog/sample.md]: Not sure about this, don't think this is needed
```suggestion
```

**Comment by ramonsmits** [comment]
Thanks @pnevmap for cloning one of the existing examples and adjusting it for datadog. We can use this if you sign the contributor license agreement.

**Comment by pnevmap** [comment]
Hi @ramonsmits, thanks for commenting and helping my with very first PR. 
The particularbot rejects the 'content-verification' and 'build-sample-and-snippets' checks but I get no feedback on what I need to adjust to pass those checks. Any advice?

**Comment by HEskandari** [comment]
@pnevmap We run a bunch of tests on the content and referred snippets (to ensure they'd compile, etc.). I had a look into that broken test. This is the error message:

```
System.AggregateException : One or more errors occurred.
  ----> System.Exception : Could not render page: C:\BuildAgent\work\a48e3341ae42e493\samples\logging\datadog\sample.md
  ----> System.Exception : Failed to render sample. PackageId:NServiceBus. Version:[7.0.0, 8.0.0).
  ----> System.Exception : Some snippets exist that were not used in markdown. Or, the snippet may be used in some versions, but not all, in which case you need to remove the snippet from the version in which it's not needed. These are the unused snippets and the versions in which they weren't used: metrics-name-mapping (>= 7.0.0 && < 8.0.0) setup-datadog-client (>= 7.0.0 && < 8.0.0) datadog-enable-nsb-metrics (>= 7.0.0 && < 8.0.0)
```

**Comment by pnevmap** [comment]
> @pnevmap We run a bunch of tests on the content and referred snippets (to ensure they'd compile, etc.). I had a look into that broken test. This is the error message:
> 
> ```
> System.AggregateException : One or more errors occurred.
>   ----> System.Exception : Could not render page: C:\BuildAgent\work\a48e3341ae42e493\samples\logging\datadog\sample.md
>   ----> System.Exception : Failed to render sample. PackageId:NServiceBus. Version:[7.0.0, 8.0.0).
>   ----> System.Exception : Some snippets exist that were not used in markdown. Or, the snippet may be used in some versions, but not all, in which case you need to remove the snippet from the version in which it's not needed. These are the unused snippets and the versions in which they weren't used: metrics-name-mapping (>= 7.0.0 && < 8.0.0) setup-datadog-client (>= 7.0.0 && < 8.0.0) datadog-enable-nsb-metrics (>= 7.0.0 && < 8.0.0)
> ```

Thanks a mil for sharing the check results @HEskandari. I was enabled to do the required changes and now the checks are green :)

**Comment by HEskandari** [comment]
@ramonsmits since you reviewed this originally, can you approve and merge?

File: samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net472</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="DogStatsD-CSharp-Client" Version="3.4.0" />
```
Comment (ramonsmits) [inline samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj]:
```suggestion
    <PackageReference Include="DogStatsD-CSharp-Client" Version="3.*" />
```

File: samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net472</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="DogStatsD-CSharp-Client" Version="3.4.0" />
+    <PackageReference Include="NServiceBus" Version="7.2.0" />
```
Comment (ramonsmits) [inline samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj]:
```suggestion
    <PackageReference Include="NServiceBus" Version="7.*" />
```

File: samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net472</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="DogStatsD-CSharp-Client" Version="3.4.0" />
+    <PackageReference Include="NServiceBus" Version="7.2.0" />
+    <PackageReference Include="NServiceBus.Metrics" Version="3.0.0" />
```
Comment (ramonsmits) [inline samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj]:
```suggestion
    <PackageReference Include="NServiceBus.Metrics" Version="3.*" />
```

File: samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net472</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="DogStatsD-CSharp-Client" Version="3.*" />
+    <PackageReference Include="NServiceBus" Version="7.2.0" />
```
Comment (ramonsmits) [inline samples/logging/datadog/Core_7/Endpoint/Endpoint.csproj]:
```suggestion
    <PackageReference Include="NServiceBus" Version="7.*" />
```

**Comment by ramonsmits** [comment]
@pnevmap please commit the suggestion. The build server is giving an unrelated content verification error.

**Comment by ramonsmits** [comment]
@pnevmap also, please rebase the PR based on master

**Pull request merged**